### ISSUE TITLE
A11y Bug: Select.Trigger does not correctly pass id to hidden select, breaking Label semantic association

### ISSUE THREAD

**Issue by JRhodes95** [radix-ui/primitives#3294]
### Current Behavior

When using the Select primitives with the Label primitive or a html label element, passing an id to the Select.Trigger and a htmlFor to the Label or label does not create a semantic link between the Label and the Select.

This throws issues in automated A11y tooling ([Wave](https://wave.webaim.org/extension/) shown in screenshots below).

![CleanShot 2024-12-17 at 15 59 03@2x](https://github.com/user-attachments/assets/ed3d15a6-eff9-40a4-b600-3a5a37b77045)
![CleanShot 2024-12-17 at 15 59 27@2x](https://github.com/user-attachments/assets/4eecfaa5-8a52-4260-9c85-024a900213c2)
![CleanShot 2024-12-17 at 15 59 49@2x](https://github.com/user-attachments/assets/801ca992-606f-43cb-b5d6-ab7f93f8f780)
![CleanShot 2024-12-17 at 15 59 57@2x](https://github.com/user-attachments/assets/7271356a-027b-4337-807b-db35e3d5ddc9)

### Expected behavior

I had expect this to create the semantic link, adding the id to the hidden select element.

### Reproducible example

[CodeSandbox repro](https://codesandbox.io/p/sandbox/gallant-glitter-g6nwg2?workspaceId=ws_MZU3G4CPtN7PLkFWz9W7Bo)
I had to install a few packages, but this is mostly just your examples from the [Select docs](https://www.radix-ui.com/primitives/docs/components/select) with one addition to allow me to pass an id to the SelectDemo component.

### Suggested solution

Make the Select.Trigger pass the id correctly?

### Additional context

I have come across similar issues before where you folks have well reasoned examples to go around the automated or MDN recommendations for semantic HTML to provide better a11y for actual cases you've tested - perhaps this is one of those?

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |   react-select      | 2.1.4        |
| React            | n/a     |   17 (your sandbox) & 18 (our design system)      |
| Browser          |    chrome     |   Version 131.0.6778.140 (Official Build) (arm64)      |
| Assistive tech   |    Wave     |         |
| Node             | n/a     |    20.8.0 (design system)     |
| npm/yarn         |    yarn (design system)     |  4.0.2        |
| Operating System |   MacOS      |     14.7.1 (23H222)    |

**Comment by JRhodes95** [comment]
Could this be related - https://github.com/radix-ui/primitives/issues/2836 ?

**Comment by Kaspazza** [comment]
Yes, id is not passed to underlying native select component, which also breaks native form behavior. Making e.g. form to be submittable without data filled in.
